source/fuzz/fuzzer_pass_add_parameters.cpp

```diff
@@ -70,14 +87,62 @@ void FuzzerPassAddParameters::Apply() {
         GetFuzzerContext()->GetRandomNumberOfNewParameters(
             GetNumberOfParameters(function));
     for (uint32_t i = 0; i < num_new_parameters; ++i) {
-      ApplyTransformation(TransformationAddParameter(
-          function.result_id(), GetFuzzerContext()->GetFreshId(),
-          // We mark the constant as irrelevant so that we can replace it with a
-          // more interesting value later.
-          FindOrCreateZeroConstant(
-              type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)],
-              true),
-          GetFuzzerContext()->GetFreshId()));
+      uint32_t current_type_id =
+          type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)];
+      auto current_type =
+          GetIRContext()->get_type_mgr()->GetType(current_type_id);
+      auto current_instr =
+          GetIRContext()->get_def_use_mgr()->GetDef(current_type_id);
+
+      if (current_type->kind() == opt::analysis::Type::kPointer) {
+        auto current_pointee_type_id =
+            fuzzerutil::GetPointeeTypeIdFromPointerType(GetIRContext(),
+                                                        current_type_id);
+        auto storage_class =
+            fuzzerutil::GetStorageClassFromPointerType(current_instr);
+
+        // Look for available variables that have the type |current_type|.
+        std::vector<uint32_t> available_variable_ids;
+        GetIRContext()->module()->ForEachInst(
+            [this, &available_variable_ids,
+             current_type_id](opt::Instruction* instruction) {
+              if (instruction->opcode() != SpvOpVariable) {
+                return;
+              }
+              if (instruction->type_id() != current_type_id) {
+                return;
+              }
+              available_variable_ids.push_back(instruction->result_id());
+            });
+        uint32_t initializer_id =
+            FindOrCreateZeroConstant(current_pointee_type_id, true);
```

Comment:
I'm not sure whether this is a good or a bad practice to mark a variable initializer as irrelevant. @paulthomson, what do you think? In case we decide to make it irrelevant, it would be good to add a comment similar to
> We mark the constant as irrelevant so that we can replace it with a more interesting value later.

Reply:
I think it works.